File: Bubble_sort.cpp
//Bubble sort
// Time Complexcity :-  O(n^2)
#include<iostream>
using namespace std;
 
void Bubble_sort(int arr[],int n)
{
    for(int i=n-1;i>=0;i--)
    {
        for(int j=0;j<=i-i;j++) // it also can be j<=i as last element should not to compare in bubble sort
        {
            if(arr[j]>arr[j+1])
            {
                int temp = arr[j+1];
                arr[j+1] = arr[j];
                arr[j] = temp;
            }
        }
    }
}


int main()
{
    int n;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++)
    cin>>arr[i];
    Bubble_sort(arr,n);
    for(int i=0; i<n;i++)
    {
        cout<<arr[i]<<" ";
    }
    return 0;
}


// output:-

// 6

// 13
// 46
// 24
// 52 
// 9
// 20
// 9 13 20 24 46 52

File: Bubble_sort.java
public class Bubble_sort
{
    public static void bubble_sort(int arr[])
    {
        for(int i=0;i<arr.length-1;i++)
        {
            int swap = 0;
            for(int j=0;j<arr.length-1-i;j++)
            {
                if(arr[j] > arr[j+1])
                {
                    int temp = arr[j];
                    arr[j] =  arr[j+1];
                    arr[j+1] = temp;
                    swap++;
                }
            }
            if (swap == 0)
            {
                break;
            }
        }
    }

    public static void bubble_sort2(int arr[])
    {
        for(int i=arr.length-1;i>=0;i--)
        {
            int swap = 0;
            for(int j=0;j<=i-1;j++)
            {
                if(arr[j] > arr[j+1])
                {
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                    swap++;
                }
            }
        if (swap==0)
        {
            break;
        }
        }
    }

    public static void printarr(int arr[])
    {
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i] + " ");
        }
    }
    public static void main (String args[])
    {
        int arr[]={2,1,5,4,3};
        bubble_sort2(arr);
        printarr(arr);
    }
}